lib/ansible/modules/windows/win_dsc.ps1

```diff
@@ -0,0 +1,236 @@
+#!powershell
+# (c) 2015, Trond Hindenes <trond@hindenes.com>, and others
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+#Temporary fix
+#Set-StrictMode -Off
+
+$params = Parse-Args $args -supports_check_mode $true
+$result = @{
+    changed = $false
+}
+
+#Check that we're on at least Powershell version 5
+if ($PSVersionTable.PSVersion.Major -lt 5)
+{
+    Fail-Json -obj $Result -message "This module only runs on Powershell version 5 or higher"
+}
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$resourcename = Get-AnsibleParam -obj $params -name "resource_name" -type "str" -failifempty $true -resultobj $result
```

Comment:
I would simply call this `$resource_name`, so that there is a 1:1 mapping between parameter and variable.
But that's just personal preference.